pages/api/extension/relevant.ts

```diff
@@ -0,0 +1,90 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { getReviewData, getFileData, getHunkData } from '../../../utils/db/relevance'
+import { getToken } from 'next-auth/jwt'
+import { getUserEmails } from '../../../utils/db/users'
+
+const getUser = async (req: NextApiRequest, res: NextApiResponse) => {
+  const user = await getToken({req: req})
+  return (user?.email) ? await getUserEmails(user.email) : new Set<string>();
+}
+
+const relevantHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  // For cors prefetch options request
+  if (req.method == "OPTIONS") {
+    res.setHeader("Access-Control-Allow-Headers", "Access-Control-Allow-Origin, Content-Type, Authorization");
+    res.status(200).send("Ok");
+    return;
+  }
+  // For normal requests
+  console.info("Getting relevant info for ", req.body.repo_name);
+  const user_emails = await getUser(req, res);
+  const { type } = req.query;
+  if (!("repo_provider" in req.body) || 
+    !("repo_owner" in req.body) || 
+    !("repo_name" in req.body)) {
+    res.status(401).send('Invalid request body');
+  }
+  let formattedData;
+  if (type === 'review') {
+      let review_db = await getReviewData(req.body.repo_provider,
+      req.body.repo_owner,
+      req.body.repo_name,
+      user_emails);
+      formattedData = await formatReviewResponse(review_db);
+  } else if (type === 'file') {
+    if (!("pr_number" in req.body)) {
+      res.status(400).send('Invalid request body');
```

Comment:
```suggestion
      res.status(400).send('Invalid request body, pr_number not present in request body');
```

Reply:
It is usually a measure of security, not giving up exactly what is wrong with the request body to the requester. If somebody is trying to impersonate and attack us, they won't know exactly what is wrong with the request, and won't be able to reverse engineer the correct request. If we want to know the exact error, we should use console.error instead of response body to specify what is wrong

Reply:
Ahh I see. This is a smart move 😅